### PR TITLE
elasticrest package added & code cleanup

### PR DESCRIPTION
**Pull request by frenchbread**
Hey @apinf/developers please review this pull request. I have pretty much cleaned up everything as well as added `elasticrest` package. Since `ApiUmbrellaElastic` class with its method has been moved to separate package, just `getMonthAnalytics` function is left in that file.

### CODE REVIEW

File: server/elasticSearch.js [hunk 1]
```diff
@@ -1,93 +1,57 @@
...
-      var values = [];     // exact values that are used in chart
-      var counts = {};     // object that handles dates and amount of requests this day
 
-      // time filter for fetching data
-      var timeFrames = {
-        start: moment().subtract(2, "weeks"),
-        end  : moment()
-      };
+    // counting total number of requests per day
+    datesArray.forEach(function(x) { counts[x] = (counts[x] || 0)+1; });
```
Comment (brylie) [inline server/elasticSearch.js]: Why do you add one here?

File: server/elasticSearch.js [hunk 2]
```diff
@@ -1,93 +1,57 @@
...
 
-      // looping items, getting the date value from an item and pushing it to datesArray
-      items.forEach(function (item) {
-        var stamp = new Date(item._source.request_at);
-        var date = stamp.getDate();
-        datesArray.push(date);
-      });
+    // looping through all the dates that are in needed range
+    var loopThroughDates = moment().range(timeFrames.start, timeFrames.end).by('days', function(moment){
```
Comment (brylie) [inline server/elasticSearch.js]: Check indentation.

File: server/elasticSearch.js [hunk 3]
```diff
@@ -1,93 +1,57 @@
...
 
+      var date = moment.date(); // variable handling current date
```
Comment (brylie) [inline server/elasticSearch.js]: How are months displayed? E.g. if the date range is from June 28th - July Nth?

File: server/elasticSearch.js (hunk 1 above)
Comment (frenchbread) [inline server/elasticSearch.js]: @brylie  It is counting function, so it increments a counter value when it finds the same value in an datesArray.

File: server/methods/elasticSearch.js [hunk 4]
```diff
@@ -1,16 +1,17 @@
 Meteor.methods({
   "getChartData": function () {
```
Comment (brylie) [inline server/methods/elasticSearch.js]: Parametrize this function to accept index, type, and limit inputs.

File: server/methods/elasticSearch.js [hunk 5]
```diff
@@ -1,16 +1,17 @@
...
-    var newSearch = new ApiUmbrellaElastic();
+
 
     // TODO: dynamic input data
-    var searchIndex = 'api-umbrella-logs-v1-2015-07';
-    var searchType = 'log';
-    var returnItemsLimit = 1000;
+    var index = 'api-umbrella-logs-v1-2015-07';
```
Comment (brylie) [inline server/methods/elasticSearch.js]: Get these values from the Method call parameters.

File: server/elasticSearch.js (hunk 1 above)
Comment (brylie) [inline server/elasticSearch.js]: OK, cool. I thought it was a data value displayed in the chart. All good.

**Comment by brylie** [comment]
Thanks for your work here and for clarifying the code with comments.

**Pull request merged**